### ISSUE TITLE
Initialize `editorState` multiple times

### ISSUE THREAD

**Issue by vishrutshah** [andrerpena/react-mde#132]
How can I initialize `editorState` multiple times based on the state changes of the parent? 

It doesn't seem to re-render with new markdown text. 

```
....

export interface IProps {
  layout: 'horizontal' | 'vertical' | 'tabbed'
  markdown: string;
}

export interface IState {
  mdeState: ReactMdeTypes.MdeState | any;
}

class MDEditor extends React.Component<IProps, IState> {
  public static defaultProps = {
    layout: 'horizontal',
    markdown: ''
  }
  private converter: Showdown.Converter;

  constructor(props: IProps) {
    super(props);
    this.state = {
      mdeState: {
        markdown: props.markdown
      },
    };
    this.converter = new Showdown.Converter({ tables: true, simplifiedAutoLink: true });
    this.handleValueChange = this.handleValueChange.bind(this);
    this.handleMarkdown = this.handleMarkdown.bind(this);
  }

  public render() {
    return (
      <div>
        <ReactMde
          layout={this.props.layout}
          onChange={this.handleValueChange}
          editorState={this.state.mdeState}
          generateMarkdownPreview={this.handleMarkdown}
        />
      </div>
    );
  }

  private handleValueChange(mdeState: ReactMdeTypes.MdeState) {
    this.setState({ mdeState });
  }

  private handleMarkdown(markdown: string): Promise<string> {
    return Promise.resolve(this.converter.makeHtml(markdown));
  }
}

export default MDEditor;

```

Calling it from other component as 

```
<MDEditor markdown={this.state.data} />
```

When data changes in `<MDEditor markdown={this.state.data} />` I would like to re-render the component `ReactMde` with new data which is not happening. I may be doing something wrong. It's be great if someone can throw some light.

**Comment by andrerpena** [comment]
Thanks for reporting and sorry that I've been away for the last couple of months. This will hopefully be fixed on version 6.0

**Comment by andrerpena** [comment]
This problem should be fixed on version 6.0.0-alpha.1. I'm sorry but this contains a number of [breaking changes](https://github.com/andrerpena/react-mde/blob/master/docs-md/ChangeLogMigrating.md#from-5-to-6). Please let me know if it does not work for you.

**Comment by andrerpena** [comment]
Now react-mde is fully controlled
